flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/ChunkedByteBuf.java

```diff
@@ -88,6 +102,16 @@ public ByteBuf readChunk(ChannelHandlerContext ctx) throws Exception {
 		}
 	}
 
+	@Override
+	public long length() {
+		return length;
+	}
+
+	@Override
+	public long progress() {
+		return length - buf.readableBytes();
```

Comment:
If the length is not know, then I think it would be better to return something like `buf.readerIndex - initialReaderIndex`.

Reply:
changed to `return buf.readerIndex()` since progress is not well defined.